decentraland/unity-explorer
# fix: make SignMessage use personal_sign to sign with wallet

## What does this PR change?
Fix #1199 

This change makes the `SignMessage` function (in the `EthereumController.js` API) request the users to sign from the browser using the [personal_sign](https://docs.metamask.io/wallet/reference/personal_sign) MetaMask operation instead of using the ephimeral key that only worked for authentications inside Decentraland.

## How to test the changes?
1. Launch the explorer.
2. `/goto BagWorlds`
3. Notice your metamask wallet is requesting you to sign through the browser.
4. Observe the error `[JAVASCRIPT]: SceneError: TypeError: (intermediate value).json is not a function` doesn't appear in the console anymore and the scene is loaded correctly.

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
